pkg/kademlia/queue.go

```diff
@@ -4,152 +4,123 @@
 package kademlia
 
 import (
-	"container/heap"
-	"math/big"
+	"math/bits"
+	"sort"
 	"sync"
 
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/storj"
 )
 
-// XorQueue is a priority queue where the priority is key XOR distance
-type XorQueue struct {
+// Queue is a limited priority queue with xor distance
+type Queue struct {
 	maxLen int
+	mu     sync.Mutex
+	added  map[storj.NodeID]int
+	items  []queueItem
+}
 
-	mu    sync.Mutex
-	added map[storj.NodeID]int
-	items items
+// An queueItem is something we manage in a priority queue.
+type queueItem struct {
+	node     *pb.Node
+	priority storj.NodeID
+	// TODO: switch to using pb.NodeAddress to avoid pointer to *pb.Node
 }
 
-// NewXorQueue returns a items with priority based on XOR from targetBytes
-func NewXorQueue(size int) *XorQueue {
-	return &XorQueue{
-		items:  make(items, 0, size),
+// xorNodeID returns the xor of each byte in NodeID
+func xorNodeID(a, b storj.NodeID) storj.NodeID {
```

Comment:
routing_helpers.go also has xorTwoIds()... maybe this should be moved into NodeId?